lib/wordy_cases.rb

```diff
@@ -0,0 +1,44 @@
+class WordyCase < OpenStruct
+  def test_name
+    'test_%s' % description.downcase.tr_s(" ", '_')
+  end
+
+  def workload
+    [
+      "question = '#{input}'",
+      indent(4, assertion),
+    ].join("\n")
+  end
+
+  def skipped
+    index.zero? ? '# skip' : 'skip'
+  end
+
+  private
+
+  def indent(size, lines)
+    lines.split("\n").map do |line|
+      ' ' * size + line
+    end.join("\n")
+  end
+
+  def assertion
+    return error_assertion unless expected
+
+    "assert_equal(#{expected}, WordProblem.new(question).answer)"
+  end
+
+  def error_assertion
+    [
+      "assert_raises ArgumentError do",
+      indent(2, "WordProblem.new(question).answer"),
+      "end",
+    ].join("\n")
+  end
+end
+
+WordyCases = proc do |data|
+  JSON.parse(data)['cases'].map.with_index do |row, i|
+    WordyCase.new(row.merge('index' => i))
```

Comment:
You can preprocess the json here to add a failure message:
```
wordy_case = WordyCase.new(row.merge('index' => i, failure_message => nil))
if wordy_case.question = 'What is -3 plus 7 multiplied by -2?'
  wordy_case.failure_message = 'You should ignore order of precedence. -3 + 7 * -2 = -8, not -17'
end
wordy_case
```
(untested, a Github comment box is not an ideal programming environment.)